### DISCUSSION TITLE
Distance matrix vs. duration matrix / edge weights

### DISCUSSION THREAD

**Discussion by bmorlo** [PyVRP/PyVRP#585] [Q&A] [answered]
Hi,

another question from my side: If I set both the distance and the duration of an edge, which of the two will the solver use for optimization? Is the duration of an edge only used in VRPTWs? I am currently thinking about relying solely on the driving duration between clients for my optimization instead of the driving distance.

Additionally, is there a way to restrict the solver to only use edges that are shorter than some value (duration)? I could probably just "not create" the edges that are longer than some value, right?

Allow me one more question: When an edge is not created, the solver internally sets it to a very high distance value, correct? Now, how can I be certain that this value is working for the distances that I am dealing with? E.g. what if I am dealing with such high distance values that the solver would actually decide to use this "not created" edge?

Is there such a "very high value" also for the durations between clients?

Thanks & cheers as always!
Benedikt

**Comment by leonlan** [comment, marked as answer]
> another question from my side: If I set both the distance and the duration of an edge, which of the two will the solver use for optimization? Is the duration of an edge only used in VRPTWs? I am currently thinking about relying solely on the driving duration between clients for my optimization instead of the driving distance.

Both distances and durations are used but in different ways.
- The distance is used to compute edge costs. This means that distances are required for optimization, i.e., minimizing the objective. It is also be used for the maximum distance constraint.
- The duration is used to calculate the time component in VRPs. This means that duration is only required if you want to satisfy temporal constraints (time windows, max duration, ...)

(Note: for PyVRP v0.9.0, we will also allow duration to become part of the objective, but this will take a few more months.) 

> Additionally, is there a way to restrict the solver to only use edges that are shorter than some value (duration)? I could probably just "not create" the edges that are longer than some value, right?

You can simply leave out those edges as pre-processing step. The solver will put a very large values on edges that are not set. 

> Allow me one more question: When an edge is not created, the solver internally sets it to a very high distance value, correct? Now, how can I be certain that this value is working for the distances that I am dealing with? E.g. what if I am dealing with such high distance values that the solver would actually decide to use this "not created" edge?

> Is there such a "very high value" also for the durations between clients?

Assuming that you use the Model interface and PyVRP v0.8.2, see code below for more details.

https://github.com/PyVRP/PyVRP/blob/991b63355e2ca5a18a87b51028172d3b0d09fc61/pyvrp/Model.py#L301-L321

https://github.com/PyVRP/PyVRP/blob/991b63355e2ca5a18a87b51028172d3b0d09fc61/pyvrp/constants.py#L1-L7
